File: Java/LegacyCodeKata/src/main/java/org/craftedsw/trip/TripDAO.java
package org.craftedsw.trip;

import java.util.List;

import org.craftedsw.exceptions.DependentClassCallDuringUnitTestException;
import org.craftedsw.user.User;



public class TripDAO {

	public static List<Trip> findTripsByUser(User user) {
		throw new DependentClassCallDuringUnitTestException(
				"TripDAO should not be invoked on an unit test.");
	}
	
	public List<Trip> findTripsBy(User user) {
		return TripDAO.findTripsByUser(user);
	}
	
}

File: C#/TripServiceLegacyKata/Trip/TripDAO.cs
using System.Collections.Generic;
using TripServiceLegacyKata.Exception;

namespace TripServiceLegacyKata.Trip
{
    public class TripDAO
    {
        public static List<Trip> FindTripsByUser(User.User user)
        {
            throw new DependendClassCallDuringUnitTestException
                ("TripDAO should not be called when running a unit test!");
        }
    }
}
